### PR TITLE
implement setAttributes:ofItemAtPath:error: method

### PR DESCRIPTION
**Pull request by qmfrederik**
Cherry-picked from the Eggplant/Keysight branch

### CODE REVIEW

**Review by rfm — APPROVED** [review]
Looks fine to me

**Comment by qmfrederik** [comment]
Thanks @rfm .  I think I need some additional approval to kick off the CI jobs? There should be an "Approve and Run" button? https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks

**Comment by qmfrederik** [comment]
The MSVC build failure seems to be related to ICU acquisition:

> Error getting latest ICU release

Is there anything you need me to do to address this?

**Comment by hmelder** [comment]
> The MSVC build failure seems to be related to ICU acquisition:

There are still some problems with the new ICU release on Windows, but @triplef pinned the current GNUstep toolchain a older release AFAIK. You can just ignore it.

**Comment by triplef** [comment]
I think this was an intermittent failure (downloading ICU failed). Re-running the CI fixed it.

File: Headers/Foundation/NSFileManager.h [hunk 1]
```diff
@@ -325,6 +325,7 @@ GS_EXPORT_CLASS
 - (BOOL) changeCurrentDirectoryPath: (NSString*)path;
 - (BOOL) changeFileAttributes: (NSDictionary*)attributes
 		       atPath: (NSString*)path;
+- (BOOL) setAttributes:(NSDictionary *)attributes ofItemAtPath:(NSString *)path error:(NSError **)error;
```
Comment (fredkiefer) [inline Headers/Foundation/NSFileManager.h:328]: The spacing here could be more consistent with the rest of the code.

File: Source/NSFileManager.m [hunk 2]
```diff
@@ -685,6 +685,14 @@ - (BOOL) changeFileAttributes: (NSDictionary*)attributes atPath: (NSString*)path
   return allOk;
 }
 
+- (BOOL) setAttributes:(NSDictionary *)attributes ofItemAtPath:(NSString *)path error:(NSError **)error
```
Comment (fredkiefer) [inline Source/NSFileManager.m:688]: Again you could try to use the spacing from the existing code

**Review by fredkiefer — APPROVED** [review]

**Comment by qmfrederik** [comment]
Thanks all for the feedback. @fredkiefer The spacing wasn't 100% consistent within the file.  Let me know if you want me to amend the spacing I ended up using.

File: Source/NSFileManager.m [hunk 3]
```diff
@@ -685,6 +685,16 @@ - (BOOL) changeFileAttributes: (NSDictionary*)attributes atPath: (NSString*)path
   return allOk;
 }
 
+- (BOOL) setAttributes: (NSDictionary *)attributes
+          ofItemAtPath: (NSString *)path
+                 error: (NSError **)error
+{
+  BOOL success = [self changeFileAttributes:attributes atPath:path];
```
Comment (fredkiefer) [inline Source/NSFileManager.m:692]: I would still add a space after the colons in this line, but the rest looks great to me.

**Comment by qmfrederik** [comment]
Thanks @fredkiefer , I made the change you requested .  Let me know if there's anything else you need!

File: Headers/Foundation/NSFileManager.h (hunk 1 above)
Comment (fredkiefer) [reply Headers/Foundation/NSFileManager.h:328]: In this file we normally have a space after the colon and none before the star (The later is not followed consistently), which actually matches the GNUstep coding style.

Comment (fredkiefer) [inline Headers/Foundation/NSFileManager.h:328]: I think this is a MasOS 10.5 method and for this reason should be moved up to line 316.

**Pull request merged**